scripts/pathseq/wdl/pathseq_pipeline.wdl

```diff
@@ -20,179 +26,712 @@
   Boolean is_host_aligned
-  Boolean? skip_quality_filters
+
+  File? kmer_file
+  File? filter_bwa_image
+  # Required if filtering_only=true
+  File? microbe_bwa_image
+  File? microbe_dict
+  File? taxonomy_file
+
+  # If enabled, filter metrics will be estimated using a downsampled bam with this many reads (recommended)
+  # If disabled, no filter metrics will be generated
+  Boolean downsample = false
+  Int downsample_reads = 1000000
+
+  # Enable to only perform host filtering
+  Boolean filtering_only = false
+
+  # Only recommended if downsample = true
+  Boolean gather_filter_metrics = false
+
+  # This can be calculated from a downsampled run to help optimize disk allocation during filtering
+  Float frac_non_host_reads = 1.0
+
+  # Filtering options
   Boolean? filter_duplicates
+  Boolean? skip_quality_filters
+  Int? host_min_identity
+  Int? host_kmer_threshold
+  Int? filter_bwa_seed_length
+  Int? min_clipped_read_length
+  Int? max_masked_bases
+  Int? min_base_quality
+  Int? quality_threshold
+  Int? dust_mask_quality
+  Int? dust_window
+  Float? dust_t
+  Int? max_adapter_mistmatches
+  Int? min_adapter_length
+  Int? filter_reads_per_partition
+  Int? filter_bam_partition_size
   Boolean? skip_pre_bwa_repartition
+
+  # Alignment options
+  Int? microbe_min_seed_length
+  Int? max_alternate_hits
+  Int? bwa_score_threshold
+
+  # Taxonomy scoring options
   Boolean? divide_by_genome_length
-  Int? filter_bwa_seed_length
+  Float? min_score_identity
+  Float? identity_margin
+  Boolean? not_normalized_by_kingdom
+  Int? score_reads_per_partition_estimate
+
+  # Runtime parameters
+  String gatk_docker
+  String samtools_docker = "biocontainers/samtools:v1.9-4-deb_cv1"
+  String genomes_in_the_cloud_docker = "broadinstitute/genomes-in-the-cloud:2.3.1-1512499786"
+  String linux_docker = "ubuntu:18.10"
+
+  File? gatk4_jar_override
+
+  Int? cram_to_bam_preemptible_attempts
+  Int? downsample_preemptible_attempts
+  Int? filter_preemptible_attempts
+  Int? align_preemptible_attempts
+  Int? score_preemptible_attempts
+  Int? process_filter_metrics_preemptible_attempts
+  Int? process_score_metrics_preemptible_attempts
+
+  Int? cram_to_bam_cpu
+  Int? filter_cpu
+  Int? align_cpu
+  Int? score_cpu
+
+  Float? cram_to_bam_mem_gb
+  Int? filter_mem_gb
+  Int? align_mem_gb
+  Int? score_mem_gb
+
+  Boolean? filter_ssd
+  Boolean? align_ssd
+  Boolean? score_ssd
+
+  Int? cram_to_bam_max_retries
+
+  # Optional input to increase all disk sizes in case of outlier sample with strange size behavior
+  Int? cram_to_bam_additional_disk_gb
+  Int? downsample_additional_disk_gb
+  Int? filter_additional_disk_gb
+  Int? align_additional_disk_gb
+  Int? score_additional_disk_gb
+
+  Boolean is_bam = basename(input_bam_or_cram, ".bam") + ".bam" == basename(input_bam_or_cram)
+
+  # Convert to BAM if we have a CRAM
+  if (!is_bam) {
+    call CramToBam {
```

Comment:
This is a source of deep sorrow.  I may ask you to help the Hellbender team validate the htsjdk cram codec.  Pathseq seems like an application with a fun assortment of weird flags and alignments.